### ISSUE TITLE
[Developer] Add tool to interactively test and regression test lexical models

### ISSUE THREAD

**Issue by mcdurdin** [keymanapp/keyman#2173]
**Is your feature request related to a problem? Please describe.**
There is no built-in tool to test a lexical model. It would be nice to have one. It would be nice if it supported interactive and automated tests, as well as recording tests for regression etc.

**Comment by mcdurdin** [comment]
Currently a lexical model must be installed into Keyman on a device to be tested. It would be very helpful to be able to select a keyboard and then test the lexical model directly within Keyman without going through the install hoops. (from #2285)

**Comment by mcdurdin** [comment]
This is not a trivial feature to implement -- there are some infrastructure changes needed to make this possible. Plus the UX may need some significant work to make this work well.

Brief notes.

```
Changes required:
1. Option to install model from model editor
2. Option to load custom model?
3. Http serve model
4. List available models in banner
5. Register model against default language 
6. Select keyboard + model


A) What do we do about model vs keyboard projects?
   1. Load the keyboard project and start debugging it
   2. Then load the model project and start debugging it
   >> This is a little obtuse.

   3. What if a model could request a keyboard to test with?

B) What about non-wordlist models?
C) Do we need to rework the model editor to have a tabbed editor just like with keyboards?
   - A details tab
   - A source tab
   - A wordlist editor
   - A compile tab

   -> wordlist editor appears if parser can find it.
```

**Comment by mcdurdin** [comment]
# Mockups for lexical model editor

## Details tab

Shows metadata extracted from .model.ts file by regex. If details cannot be safely edited due to non-matching regex, then we'll make this read-only.

![image](https://user-images.githubusercontent.com/4498365/69387068-c3963a00-0d18-11ea-9389-5672b2138da9.png)

## Wordlist tab

Shows existing wordlist editor. May be multiple wordlist tabs if multiple files are referred to.

## Source tab

Shows source of .model.ts in text editor.

## Build tab

Allows for compiling the lexical model and starting the test phase.

![image](https://user-images.githubusercontent.com/4498365/69387083-ca24b180-0d18-11ea-9897-92aa44b138ac.png)

Once this is in place, we should probably stop showing the wordlist as a separate file in the project (the template will be updated to not include the wordlist in the project).
